Mathlib/Analysis/Discrete/MarcinkiewiczZygmund.lean

```diff
@@ -0,0 +1,290 @@
+/-
+Copyright (c) 2023 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Algebra.Order.Chebyshev
+import Mathlib.Analysis.InnerProductSpace.PiL2
+import Mathlib.Data.Nat.Choose.Multinomial
+import Mathlib.Tactic.Positivity.Finset
+
+/-!
+# The Marcinkiewicz-Zygmund inequality
+
+This file proves the Marcinkiewicz-Zygmund inequality. This is the statement that, for a random
+variable `f` valued in a real inner product space, the $$L^p$$-norm of the sum of `n` samples of `f`
+is bounded by the $$L^p$$-norm of `f` up to some multiplicative constant.
+
+## TODO
+
+* We currently only prove the inequality for `p = 2 * m` an even natural number. The general `p`
+  case can be obtained from this specific one by nesting of Lp norms.
+* We currently only prove the case of a uniformly distributed discrete random variable. This needs
+  to be generalised! (and when it is, this file should move out of the `Analysis.Discrete`) folder.
+-/
+
+open Finset Fintype Nat Real
+variable {α : Type*} {n : Type*} [Fintype n] {A : Finset α} {m : ℕ}
+
+private lemma step_seven {f : α → ℝ} {a b : n → α} :
```

Comment:
To help your reader follow the argument, it would help a lot to have a short proof sketch in the file-level docstring, mentioning where the various auxiliary lemmas are relevant.